src/browser/BLECentralPlugin.js

```diff
@@ -1,78 +1,220 @@
+                success(result);
+            }).catch(error => {
+                if (failure) failure(error);
+            });
+        } else if (failure) { 
+            failure(new Error("device not connected"));
+        }
+    },
+    startNotification: function(deviceId, service_uuid, characteristic_uuid, success, failure) {
+         if (this.deviceInfos.has(deviceId)) {
+            this.deviceInfos.get(deviceId).server.getPrimaryService(formatUUID(service_uuid)).then(service => {
+                return service.getCharacteristic(formatUUID(characteristic_uuid));
+            }).then(characteristic => {
+                return characteristic.startNotifications().then(result => {
+                    characteristic.addEventListener('characteristicvaluechanged', function (event) {
+                        success(event.target.value.buffer);
+                    });
+                });
+            }).catch(error => {
+                if (failure) failure(error);
+            })
+        } else if (failure) { 
+            failure(new Error("device not connected"));
+        }
+    },
+    stopNotifcation: function(deviceId, service_uuid, characteristic_uuid, success, failure) {
+       if (this.deviceInfos.has(deviceId)) {
+            this.deviceInfos.get(deviceId).server.getPrimaryService(formatUUID(service_uuid)).then(service => {
+                return service.getCharacteristic(formatUUID(characteristic_uuid));
+            }).then(characteristic => {
+                return characteristic.stopNotifications();
+            }).then(result => {
+                success(result);
+            }).catch(error => {
+                if (failure) failure(error);
+            });
+        } else if (failure) { 
+            failure(new Error("device not connected"));
+        }
     },
     isEnabled: function(success, failure) {
         notSupported();
-        if (failure) failure();
+        if (failure) failure(new Error("not supported"));
     },
-    isConnected: function(device_id, success, failure) {
-        notSupported();
-        if (failure) failure();
+    isConnected: function(deviceId, success, failure) {
+        if (this.deviceInfos.has(deviceId)) {
+            var device = this.deviceInfos.get(deviceId).device;
+            if (device.gatt.connected) {
+                success();
+            } else {
+                if (failure) failure();
+            }
+        } else if (failure) {
+            failure();
+        }
     },
     showBluetoothSettings: function(success, failure) {
         notSupported();
-        if (failure) failure();
+        if (failure) failure(new Error("not supported"));
     },
     enable: function(success, failure) {
         notSupported();
-        if (failure) failure();
+        if (failure) failure(new Error("not supported"));
     },
     startStateNotifications: function(success, failure) {
-      notSupported();
-      if (failure) failure();
+        notSupported();
```

Comment:
Just in case you weren't aware, there looks to be support for this in the web Bluetooth standard as well: https://developer.mozilla.org/en-US/docs/Web/API/BluetoothRemoteGATTCharacteristic/startNotifications

But... I'm very happy to merge this without this feature, as what is here is obviously very useful on its own

Reply:
I'm using `startNotification` successfully. Not really sure about `startStateNotifications` that function looks different to me.